logstash-core/src/main/java/org/logstash/execution/AbstractPipelineExt.java

```diff
@@ -447,6 +454,102 @@ public final IRubyObject collectStats(final ThreadContext context) throws IOExce
         return context.nil;
     }
 
+    @JRubyMethod(name = "initialize_flow_metrics")
+    public final IRubyObject initializeFlowMetrics(final ThreadContext context) {
+        final UptimeMetric uptimeInMillis = initOrGetUptimeMetric(context, buildNamespace(), context.runtime.newSymbol("uptime_in_millis"));
+        final UptimeMetric uptimeInSeconds = uptimeInMillis.withTimeUnit("uptime_in_seconds", TimeUnit.SECONDS);
```

Comment:
This approach is problematic because `UptimeMetric` is a `Metric<Long>` so we lose a significant amount of precision. If our capture cadence is 4.999s apart, the value available as the denominator to our flow metric is `4`, causing rates to be over-estimated by ~25%. I'll think through a refactor that lets us maintain precision, which may be either changing `UptimeMetric` to be `Metric<Number>` or introducing a `ScaledMetric`.